File: mycuration/src/main/java/com/phicdy/mycuration/util/UrlUtil.java
package com.phicdy.mycuration.util;

import java.util.regex.Matcher;
import java.util.regex.Pattern;


public class UrlUtil {

	public static boolean hasParameterUrl(String url) {
		String regex = "^(http|https):\\/\\/.+\\/\\?.+$";
		Pattern p = Pattern.compile(regex);
		Matcher m = p.matcher(url);
		return m.find();
	}
	
	public static String removeUrlParameter(String url) {
		if (!isCorrectUrl(url)) {
			return url;
		}
		if (!url.contains("?")) {
			return url;
		}
		return url.substring(0, url.indexOf("?"));
	}
	
	public static boolean isCorrectUrl(String url) {
		return url != null && (url.startsWith("http://") || url.startsWith("https://"));
	}
}

File: mycuration/src/main/java/com/phicdy/mycuration/util/UrlUtil.kt
package com.phicdy.mycuration.util

import java.util.regex.Pattern


object UrlUtil {

    fun hasParameterUrl(url: String): Boolean {
        val regex = "^(http|https):\\/\\/.+\\/\\?.+$"
        val p = Pattern.compile(regex)
        val m = p.matcher(url)
        return m.find()
    }

    fun removeUrlParameter(url: String): String? {
        if (!isCorrectUrl(url)) {
            return url
        }
        return if (!url.contains("?")) {
            url
        } else url.substring(0, url.indexOf("?"))
    }

    fun isCorrectUrl(url: String?): Boolean {
        return url != null && (url.startsWith("http://") || url.startsWith("https://"))
    }
}
